### PR TITLE
Reorder galleryHasMedias on postUpdate

### LINKED ISSUE #1539: GalleryHasMedia collection not sorted after update of gallery
#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.42.2 3.45.1 The missing Symfony Admin Generator
sonata-project/block-bundle              3.13.0 3.14.0 Symfony SonataBlockBundle
sonata-project/cache                     2.0.1  2.0.1  Cache library
sonata-project/core-bundle               3.12.0 3.15.0 Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.4.0  2.4.0  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.1.2  1.1.5  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.6.3  3.8.1  Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.5.0  2.5.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.9.1  2.0.0  Lightweight Exporter library
sonata-project/media-bundle              3.16.3 3.18.1 Symfony SonataMediaBundle
sonata-project/user-bundle               4.2.3  4.2.3  Symfony SonataUserBundle
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
symfony/monolog-bundle     v3.3.1  v3.3.1  Symfony MonologBundle
symfony/phpunit-bridge     v3.4.20 v4.2.2  Symfony PHPUnit Bridge
symfony/polyfill-apcu      v1.10.0 v1.10.0 Symfony polyfill backporting apcu_* functions to lower PHP versions
symfony/polyfill-ctype     v1.10.0 v1.10.0 Symfony polyfill for ctype functions
symfony/polyfill-intl-icu  v1.10.0 v1.10.0 Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.10.0 v1.10.0 Symfony polyfill for the Mbstring extension
symfony/polyfill-php56     v1.10.0 v1.10.0 Symfony polyfill backporting some PHP 5.6+ features to lower PHP versions
symfony/polyfill-php70     v1.10.0 v1.10.0 Symfony polyfill backporting some PHP 7.0+ features to lower PHP versions
symfony/polyfill-php72     v1.10.0 v1.10.0 Symfony polyfill backporting some PHP 7.2+ features to lower PHP versions
symfony/polyfill-util      v1.10.0 v1.10.0 Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.1  v3.0.1  Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v3.2.4  v3.2.5  Symfony SwiftmailerBundle
symfony/symfony            v3.4.20 v4.2.2  The Symfony PHP framework
```

#### PHP version

```
$ php -v
PHP 7.2.10-0ubuntu0.18.04.1 (cli) (built: Sep 13 2018 13:45:02) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.10-0ubuntu0.18.04.1, Copyright (c) 1999-2018, by Zend Technologies

```

## Subject

When I update a media position on a gallery and I add code in the `postUpdate` admin method, I receive a `$gallery` instance with a `galleryHasMedias` collection. This collection is not sorted by position (It's the case when fetched from DB thanks to the orderBy doctrine's mapping).

## Steps to reproduce
- Open the gallery edit page
- Change position of a media
- Update the gallery.

## Expected results
- The `$gallery` instance passed in `postUpdate` method should be sorted.
## Actual results
It's not the case

### PR DESCRIPTION
**Pull request by tifabien**
## Subject

I am targeting this branch, but I don't know if it's consider as a BC break.

Closes #1539 

## Changelog

```markdown
### Changed
- Sort galleryHasMedias on GalleryAdmin::postUpdate
```

### CODE REVIEW

File: src/Admin/GalleryAdmin.php [hunk 1]
```diff
@@ -53,6 +53,23 @@ public function prePersist($gallery)
         $gallery->setContext($parameters['context']);
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function postUpdate($gallery)
+    {
+        if (false === $gallery->getGalleryHasMedias() instanceof \IteratorAggregate) {
```
Comment (greg0ire) [inline src/Admin/GalleryAdmin.php]:
```suggestion
        if (!$gallery->getGalleryHasMedias() instanceof \IteratorAggregate) {
```

File: src/Admin/GalleryAdmin.php [hunk 2]
```diff
@@ -53,6 +53,23 @@ public function prePersist($gallery)
...
+            return;
+        }
+
+        $iterator = $gallery->getGalleryHasMedias()->getIterator();
+        $iterator->uasort(function ($a, $b) {
```
Comment (greg0ire) [inline src/Admin/GalleryAdmin.php]: parameter and return type hints

File: src/Admin/GalleryAdmin.php [hunk 3]
```diff
@@ -53,6 +53,23 @@ public function prePersist($gallery)
...
+            return ($a->getPosition() < $b->getPosition()) ? -1 : 1;
```
Comment (greg0ire) [inline src/Admin/GalleryAdmin.php]:
```suggestion
            return $a->getPosition() <=> $b->getPosition();
```

**Comment by greg0ire** [comment]
Why not call https://github.com/sonata-project/SonataMediaBundle/blob/edabd96f4ab83d75c08c32620d8ca7d07d0b6c37/src/PHPCR/BaseGallery.php#L89-L106 ?

**Comment by OskarStark** [comment]
Because it’s from PHPCR namespace

**Comment by greg0ire** [comment]
Oh right… maybe we could just move it in the parent class?

**Pull request merged**